From a1eac8a67077ee7f99a49c9358c4df5ae4f5ab9a Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Mon, 27 Oct 2025 15:37:17 +0100
Subject: [PATCH] point to our own llvmdev

---
 tinygrad/runtime/support/llvm.py | 26 ++++++++++++++++++--------
 1 file changed, 18 insertions(+), 8 deletions(-)

diff --git a/tinygrad/runtime/support/llvm.py b/tinygrad/runtime/support/llvm.py
index bb1490e58..a14391356 100644
--- a/tinygrad/runtime/support/llvm.py
+++ b/tinygrad/runtime/support/llvm.py
@@ -1,24 +1,34 @@
 import ctypes, ctypes.util, os, sys, subprocess
 from tinygrad.helpers import DEBUG, OSX, getenv
 
+CONDA_PREFIX = os.environ.get('CONDA_PREFIX', None)
+
 if sys.platform == 'win32':
   # Windows llvm distribution doesn't seem to add itself to PATH or anywhere else where it can be easily retrieved from.
   # winget also doesn't have something like `brew --prefix llvm` so just hardcode default installation path with an option to override
-  LLVM_PATH = getenv('LLVM_PATH', 'C:\\Program Files\\LLVM\\bin\\LLVM-C.dll')
+  if CONDA_PREFIX is not None:
+    LLVM_PATH = os.path.join(CONDA_PREFIX, "Library", "bin", "LLVM-C.dll")
+  else:
+    LLVM_PATH = getenv('LLVM_PATH', 'C:\\Program Files\\LLVM\\bin\\LLVM-C.dll')
   if not os.path.exists(LLVM_PATH):
     raise FileNotFoundError('LLVM not found, you can install it with `winget install LLVM.LLVM` or point at a custom dll with LLVM_PATH')
 elif OSX:
+  if CONDA_PREFIX is not None:
+    prefix = CONDA_PREFIX
   # Will raise FileNotFoundError if brew is not installed
   # `brew --prefix` will return even if formula is not installed
-  if not os.path.exists(brew_prefix:=subprocess.check_output(['brew', '--prefix', 'llvm@19']).decode().strip()):
+  elif not os.path.exists(prefix:=subprocess.check_output(['brew', '--prefix', 'llvm@19']).decode().strip()):
     raise FileNotFoundError('LLVM not found, you can install it with `brew install llvm@19`')
-  LLVM_PATH: str|None = os.path.join(brew_prefix, 'lib', 'libLLVM.dylib')
+  LLVM_PATH: str|None = os.path.join(prefix, 'lib', 'libLLVM.dylib')
 else:
-  LLVM_PATH = ctypes.util.find_library('LLVM')
-  # use newer LLVM if possible
-  for ver in reversed(range(14, 19+1)):
-    if LLVM_PATH is not None: break
-    LLVM_PATH = ctypes.util.find_library(f'LLVM-{ver}')
+  if CONDA_PREFIX is not None:
+    LLVM_PATH = os.path.join(CONDA_PREFIX, 'lib', 'libLLVM.so')
+  else:
+    LLVM_PATH = ctypes.util.find_library('LLVM')
+    # use newer LLVM if possible
+    for ver in reversed(range(14, 19+1)):
+      if LLVM_PATH is not None: break
+      LLVM_PATH = ctypes.util.find_library(f'LLVM-{ver}')
   if LLVM_PATH is None:
     raise FileNotFoundError("No LLVM library found on the system. Install it via your distro's package manager and ensure it's findable as 'LLVM'")
 
