From 94e617644402d12073fc6d55128a08eed9a0b7d6 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Mon, 27 Oct 2025 15:37:17 +0100
Subject: [PATCH 1/2] point to our own llvmdev

---
 tinygrad/runtime/autogen/__init__.py | 28 ++++++++++++++++++++++------
 1 file changed, 22 insertions(+), 6 deletions(-)

diff --git a/tinygrad/runtime/autogen/__init__.py b/tinygrad/runtime/autogen/__init__.py
index dc717e0d5..bac164a56 100644
--- a/tinygrad/runtime/autogen/__init__.py
+++ b/tinygrad/runtime/autogen/__init__.py
@@ -1,5 +1,11 @@
 import glob, importlib, pathlib, subprocess, tarfile
-from tinygrad.helpers import fetch, flatten, system, getenv
+from tinygrad.helpers import fetch, flatten, system, getenv, OSX, WIN
+import os
+
+CONDA_PREFIX = os.environ.get('CONDA_PREFIX', None)
+if CONDA_PREFIX is not None:
+  # if present, turn into full path; windows has a different layout
+  CONDA_PREFIX = pathlib.Path(CONDA_PREFIX)/"Library" if WIN else pathlib.Path(CONDA_PREFIX)
 
 root = (here:=pathlib.Path(__file__).parent).parents[2]
 nv_src = {"nv_570": "https://github.com/NVIDIA/open-gpu-kernel-modules/archive/81fe4fb417c8ac3b9bdcc1d56827d116743892a5.tar.gz",
@@ -79,8 +85,13 @@ def __getattr__(nm):
                                  rules=[('__NR', 'NR')])
     case "ib": return load("ib", "'ibverbs'", ["/usr/include/infiniband/verbs.h", "/usr/include/infiniband/verbs_api.h",
                                                "/usr/include/infiniband/ib_user_ioctl_verbs.h","/usr/include/rdma/ib_user_verbs.h"], errno=True)
-    case "llvm": return load("llvm", llvm_lib, lambda: [system("llvm-config-20 --includedir")+"/llvm-c/**/*.h"],
-                             args=lambda: system("llvm-config-20 --cflags").split(), recsym=True, prolog=["from tinygrad.helpers import WIN, OSX"])
+    case "llvm":
+      if CONDA_PREFIX is None:
+        return load("llvm", llvm_lib, lambda: [system("llvm-config-20 --includedir")+"/llvm-c/**/*.h"],
+                    args=lambda: system("llvm-config-20 --cflags").split(), recsym=True, prolog=["from tinygrad.helpers import WIN, OSX"])
+      else:
+        llvm_lib = CONDA_PREFIX/(r"bin\LLVM-C.dll" if WIN else f"lib/libLLVM.{'dylib' if OSX else 'so'}")
+        return load("llvm", llvm_lib, CONDA_PREFIX/"include"/"llvm-c/**/*.h", args=lambda: system("llvm-config --cflags").split(), recsym=True)
     case "pci": return load("pci", None, ["/usr/include/linux/pci_regs.h"])
     case "vfio": return load("vfio", None, ["/usr/include/linux/vfio.h"])
     # could add rule: WGPU_COMMA -> ','
@@ -132,9 +143,14 @@ def __getattr__(nm):
   tarball="https://gitlab.freedesktop.org/mesa/mesa/-/archive/mesa-25.2.7/mesa-25.2.7.tar.gz",
   prolog=["import gzip, base64"], epilog=lambda path: [system(f"{root}/extra/mesa/lvp_nir_options.sh {path}")])
     case "libclang":
-      return load("libclang", "'clang-20'",
-                  lambda: [f"{system('llvm-config-20 --includedir')}/clang-c/{s}.h" for s in ["Index", "CXString", "CXSourceLocation", "CXFile"]],
-                  args=lambda: system("llvm-config-20 --cflags").split())
+      if CONDA_PREFIX is None:
+        return load("libclang", "'clang-20'",
+                    lambda: [f"{system('llvm-config-20 --includedir')}/clang-c/{s}.h" for s in ["Index", "CXString", "CXSourceLocation", "CXFile"]],
+                    args=lambda: system("llvm-config-20 --cflags").split())
+      else:
+        return load("libclang", "'clang'",
+                    lambda: [CONDA_PREFIX/f"include/clang-c/{s}.h" for s in ["Index", "CXString", "CXSourceLocation", "CXFile"]],
+                    args=lambda: system("llvm-config --cflags").split())
     case "metal":
       return load("metal", "'Metal'", [f"{macossdk}/System/Library/Frameworks/Metal.framework/Headers/MTL{s}.h" for s in
                   ["ComputeCommandEncoder", "ComputePipeline", "CommandQueue", "Device", "IndirectCommandBuffer", "Resource", "CommandEncoder"]],
