From 1307846a23c28258a6be2e43b05b843e4d5f911d Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Sun, 28 Dec 2025 13:56:39 +1100
Subject: [PATCH 4/4] look for libraries in PREFIX first

---
 tinygrad/runtime/support/c.py | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/tinygrad/runtime/support/c.py b/tinygrad/runtime/support/c.py
index 9f99e783a..dc474dcd4 100644
--- a/tinygrad/runtime/support/c.py
+++ b/tinygrad/runtime/support/c.py
@@ -37,13 +37,18 @@ def CEnum(typ: type[ctypes._SimpleCData]):
 
   return _CEnum
 
+CONDA_PREFIX = os.environ.get('CONDA_PREFIX', None)
+# turn into full path; windows has a different layout
+CONDA_PREFIX = pathlib.Path(CONDA_PREFIX)/"Library" if WIN else pathlib.Path(CONDA_PREFIX)
+
 class DLL(ctypes.CDLL):
   @staticmethod
   def findlib(nm:str, paths:list[str], extra_paths=[]):
     if nm == 'libc' and OSX: return '/usr/lib/libc.dylib'
     if pathlib.Path(path:=getenv(nm.replace('-', '_').upper()+"_PATH", '')).is_file(): return path
     for p in paths:
-      libpaths = {"posix": ["/usr/lib", "/usr/local/lib"], "nt": os.environ['PATH'].split(os.pathsep),
+      libpaths = {"posix": [CONDA_PREFIX/"lib", "/usr/lib", "/usr/local/lib"],
+                  "nt": [CONDA_PREFIX/"bin"] + os.environ['PATH'].split(os.pathsep),
                   "darwin": ["/opt/homebrew/lib", f"/System/Library/Frameworks/{p}.framework"],
                   'linux': ['/lib', f"/lib/{sysconfig.get_config_var('MULTIARCH')}", "/usr/lib/wsl/lib/"]}
       if (pth:=pathlib.Path(p)).is_absolute():
