From 3b97d335ad3cb8e44649bd009fc7517c92f3ab19 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Thu, 25 Dec 2025 15:03:07 +1100
Subject: [PATCH 6/7] do not JIT test accuracy on osx

---
 examples/beautiful_mnist.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/examples/beautiful_mnist.py b/examples/beautiful_mnist.py
index ec0cb919f..8683e36c3 100644
--- a/examples/beautiful_mnist.py
+++ b/examples/beautiful_mnist.py
@@ -31,7 +31,6 @@ if __name__ == "__main__":
     loss = model(X_train[samples]).sparse_categorical_crossentropy(Y_train[samples]).backward()
     return loss.realize(*opt.schedule_step())
 
-  @TinyJit
   def get_test_acc() -> Tensor: return (model(X_test).argmax(axis=1) == Y_test).mean()*100
 
   test_acc = float('nan')
