From 0557eb24e313c23134f63e3fb55fba218707e717 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Sun, 28 Dec 2025 14:44:16 +1100
Subject: [PATCH 7/7] debug

---
 tinygrad/runtime/support/c.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/tinygrad/runtime/support/c.py b/tinygrad/runtime/support/c.py
index dc474dcd4..c4a078369 100644
--- a/tinygrad/runtime/support/c.py
+++ b/tinygrad/runtime/support/c.py
@@ -44,8 +44,11 @@ CONDA_PREFIX = pathlib.Path(CONDA_PREFIX)/"Library" if WIN else pathlib.Path(CON
 class DLL(ctypes.CDLL):
   @staticmethod
   def findlib(nm:str, paths:list[str], extra_paths=[]):
+    print(f"{nm=}, {paths=}, {extra_paths=}")
     if nm == 'libc' and OSX: return '/usr/lib/libc.dylib'
-    if pathlib.Path(path:=getenv(nm.replace('-', '_').upper()+"_PATH", '')).is_file(): return path
+    if pathlib.Path(path:=getenv(nm.replace('-', '_').upper()+"_PATH", '')).is_file():
+      print(f"found {path}")
+      return path
     for p in paths:
       libpaths = {"posix": [CONDA_PREFIX/"lib", "/usr/lib", "/usr/local/lib"],
                   "nt": [CONDA_PREFIX/"bin"] + os.environ['PATH'].split(os.pathsep),
@@ -61,6 +64,7 @@ class DLL(ctypes.CDLL):
             if (l:=pre / base).is_file() or (OSX and 'framework' in str(l) and l.is_symlink()): return str(l)
         else:
           for l in (l for l in pre.iterdir() if l.is_file() and re.fullmatch(f"lib{p}\\.so\\.?[0-9]*", l.name)):
+            print(f"considering file {l}")
             # filter out linker scripts
             with open(l, 'rb') as f:
               if f.read(4) == b'\x7FELF': return str(l)
